cpp/src/Slice/Util.h

```diff
@@ -8,6 +8,16 @@
 #include <Slice/Parser.h>
 #include <IceUtil/OutputUtil.h>
 
+#ifdef _MSC_VER
```

Comment:
These are needed for here in SliceUtil.cpp:
```
            prefix3 += ::tolower(static_cast<unsigned char>(name[0]));
            prefix3 += ::tolower(static_cast<unsigned char>(name[1]));
            prefix3 += ::tolower(static_cast<unsigned char>(name[2]));
```
Without them the compiler complains about lossy conversions.
```
src/Slice/SliceUtil.cpp:514:24: error: implicit conversion loses integer precision: 'int' to 'std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> >::value_type' (aka 'char') [-Werror,-Wimplicit-int-conversion]
            prefix3 += ::tolower(static_cast<unsigned char>(name[0]));
                    ~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/Slice/SliceUtil.cpp:515:24: error: implicit conversion loses integer precision: 'int' to 'std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> >::value_type' (aka 'char') [-Werror,-Wimplicit-int-conversion]
            prefix3 += ::tolower(static_cast<unsigned char>(name[1]));
                    ~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/Slice/SliceUtil.cpp:516:24: error: implicit conversion loses integer precision: 'int' to 'std::__1::basic_string<char,
      std::__1::char_traits<char>, std::__1::allocator<char> >::value_type' (aka 'char') [-Werror,-Wimplicit-int-conversion]
            prefix3 += ::tolower(static_cast<unsigned char>(name[2]));
                    ~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

`::toLower` takes and returns ints. I considered implementing our own version of it, but decided it was outside the scope of the PR and possibly overkill.

In hindsight the first `#   pragma warning(disable:4244)` probably isn't necessary anymore though. It's from earlier code I believe.

Reply:
You could add the suppress warning to the .cpp file instead of the .h file. Or simpler, use `IceUtilInternal::toLower` to perform this case-insensitive string comparison.

Reply:
Or I could use the new `ciequals` Jose added that I didn't notice :)